lib/bdev/nvme/blockdev_nvme.c

```diff
@@ -969,6 +978,36 @@ bdev_nvme_queued_done(void *ref, const struct spdk_nvme_cpl *cpl)
 }
 
 static void
+bdev_nvme_passthru_completion(void *arg1, void *arg2)
+{
+	struct spdk_bdev_io *bdev_io = arg1;
+	enum spdk_bdev_io_status status = (enum spdk_bdev_io_status)arg2;
+
+	spdk_bdev_io_complete(bdev_io, status);
+}
+
+static void
+bdev_nvme_passthru_done(void *ref, const struct spdk_nvme_cpl *cpl)
+{
+	struct spdk_bdev_io *bdev_io = spdk_bdev_io_from_ctx((struct nvme_bdev_io *)ref);
+	int sct = cpl->status.sct;
+	int sc = cpl->status.sc;
+
+	if (sct == SPDK_NVME_SCT_GENERIC && sc == SPDK_NVME_SC_SUCCESS) {
+		bdev_io->status = SPDK_BDEV_IO_STATUS_SUCCESS;
+	} else {
+		bdev_io->error.nvme.sct = sct;
+		bdev_io->error.nvme.sc = sc;
+		bdev_io->status = SPDK_BDEV_IO_STATUS_NVME_ERROR;
+	}
+
+	spdk_event_call(spdk_event_allocate(bdev_io->u.passthru.p_lcore,
+					    (spdk_event_fn)bdev_nvme_passthru_completion, bdev_io,
+					    (void *)bdev_io->status));
```

Comment:
It might be easier to just preserve the entire spdk_nvme_cpl in the bdev_io union and then call spdk_bdev_io_complete_nvme_status() in the event function.